src/OpenTelemetry.AutoInstrumentation/Configurations/EnvironmentConfigurationTracerHelper.cs

```diff
@@ -241,5 +242,13 @@ public static TracerProviderBuilder AddOtlpExporter(TracerProviderBuilder builde
                 pluginManager.ConfigureTracesOptions(options);
             });
         }
+
+        [MethodImpl(MethodImplOptions.NoInlining)]
+        public static TracerProviderBuilder AddAzureInstrumentation(TracerProviderBuilder builder)
+        {
+            const string AzureActivitySourceName = "AZURE_EXPERIMENTAL_ENABLE_ACTIVITY_SOURCE";
+            Environment.SetEnvironmentVariable(AzureActivitySourceName, "true");
```

Comment:
Is this call is _always_ early enough to be fetched by Azure SDK?
What is the better option setting this env. variable or calling method https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/core/Azure.Core/samples/Diagnostics.md?plain=1#L184-L185?

Reply:
Yes, this is a good place to enable OpenTelemetry based spans from Azure SDK. As we 
 are in the context of code, I recommend to use `AppContext.SetSwitch("Azure.Experimental.EnableActivitySource", true);`

Reply:
Fixed to use `AppContext.SetSwitch("Azure.Experimental.EnableActivitySource", true);`.